cms/private_media/wagtail_hooks.py

```diff
@@ -0,0 +1,22 @@
+from typing import TYPE_CHECKING
+
+from django.core.exceptions import PermissionDenied
+from wagtail import hooks
+from wagtail.documents.permissions import permission_policy
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest
+
+    from cms.private_media.models import PrivateDocumentMixin
+
+
+@hooks.register("before_serve_document")
+def protect_private_documents(document: "PrivateDocumentMixin", request: "HttpRequest") -> None:
+    """Block access to private documents if the user has insufficient permissions."""
+    if document.is_private and (
+        not request.user.is_authenticated
+        or not permission_policy.user_has_any_permission_for_instance(
+            request.user, ["choose", "add", "change"], document
+        )
+    ):
```

Comment:
Thought: This logic is duplicated in a few places. Perhaps it should be a method on the mixin? Something like `user_can_access` or alike?

Reply:
I don't thing that's worth the added complexity to be honest. I think we'll eventually be overriding the permission policy, and can look at it at that point. For now, we've agreed that this PR should just use stock permissions stuff.